### PR TITLE
GetLinksCommand; Services; Installed GuzzleHttp

### PR DESCRIPTION
**Pull request by DariusPa**

### CODE REVIEW

File: src/ImportBundle/Command/GetLinksCommand.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    {
+
+        $id = $input->getArgument('id');
+
+        $service = $this->getContainer()->get('import.link.parser');
+        $service2 = $this->getContainer()->get('import.product.link');
+
+        $shopLink = $service->start($id);
+
+        $client = new GuzzleHttp\Client();
```
Comment (tfe2012) [inline src/ImportBundle/Command/GetLinksCommand.php]: use GuzzleHttp\Client

File: src/ImportBundle/Command/GetLinksCommand.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                    $output->writeln($message);
+                }
+            }
+        }
+
+    }
+
+    protected function getCategoriesLinks( Crawler $crawler ) {
+
+        //$links = $crawler->filter( 'div#menu_oc ul li a' )->each( function ( Crawler $node, $i ) {
```
Comment (tfe2012) [inline src/ImportBundle/Command/GetLinksCommand.php]: remove unused code

File: src/ImportBundle/Services/LinkParser.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<?php
+namespace ImportBundle\Services;
+
+class LinkParser
```
Comment (tfe2012) [inline src/ImportBundle/Services/LinkParser.php:4]: is it final implementation?

Comment (DariusPa) [inline src/ImportBundle/Services/LinkParser.php:4]: I haven't written code for checking if link already exists in db yet.

File: src/ImportBundle/Command/GetLinksCommand.php (hunk 2 above)
Comment (DariusPa) [inline src/ImportBundle/Command/GetLinksCommand.php]: done

File: src/ImportBundle/Command/GetLinksCommand.php (hunk 1 above)
Comment (DariusPa) [inline src/ImportBundle/Command/GetLinksCommand.php]: done

File: src/ImportBundle/Command/GetLinksCommand.php [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        $links = $crawler->filter( 'div#menu_oc > ul > li > a' )->each( function ( Crawler $node, $i ) {
+            return $node->link()->getUri();
+        });
+
+        return array_values( $links );
+    }
+
+    protected function getCategoryProducts( Crawler $crawler ) {
+
+        $links = $crawler->filter( '#content div.image > a' )->each( function ( Crawler $node, $i ) {
```
Comment (tfe2012) [inline src/ImportBundle/Command/GetLinksCommand.php]: is it code universal for all pages?

File: src/ImportBundle/Command/GetLinksCommand.php [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                }
+            }
+        }
+
+    }
+
+    protected function getCategoriesLinks( Crawler $crawler ) {
+
+        //$links = $crawler->filter( 'div#menu_oc ul li a' )->each( function ( Crawler $node, $i ) {
+        $links = $crawler->filter( 'div#menu_oc > ul > li > a' )->each( function ( Crawler $node, $i ) {
```
Comment (tfe2012) [inline src/ImportBundle/Command/GetLinksCommand.php]: is it code universal for all pages?

File: src/ImportBundle/Command/GetLinksCommand.php [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+        $links = $crawler->filter( '#content div.image > a' )->each( function ( Crawler $node, $i ) {
+            return $node->link()->getUri();
+        });
+
+        return array_values( $links );
+    }
+
+    protected function getPagesCount( Crawler $crawler ) {
+        $pages = $crawler->filter( 'div.pagination div.results' )->text();
+        $result = explode("(", $pages);
```
Comment (tfe2012) [inline src/ImportBundle/Command/GetLinksCommand.php]: is it code universal for all pages?

File: src/ImportBundle/Command/GetLinksCommand.php [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        $response = $client->request('GET', $shopLink);
+        $crawler = new Crawler($response->getBody()->getContents(), 'http');
+        $categoriesLinks = $this->getCategoriesLinks($crawler);
+        foreach($categoriesLinks as $item) {
+            $response = $client->request('GET', $item);
+            $crawler = new Crawler($response->getBody()->getContents(), 'http');
+            $pagination = $this->getPagesCount($crawler);
+            for ($i = 1; $i <= $pagination; $i++) {
+                $response = $client->request('GET', $item . '/?page=' . $i);
+                $crawler = new Crawler($response->getBody()->getContents(), 'http');
```
Comment (tfe2012) [inline src/ImportBundle/Command/GetLinksCommand.php]: This 2 lines must have dependencies from interface, not from class

File: src/ImportBundle/Command/GetLinksCommand.php [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+        $shopLink = $service->start($id);
+
+        $client = new GuzzleHttp\Client();
+        $response = $client->request('GET', $shopLink);
+        $crawler = new Crawler($response->getBody()->getContents(), 'http');
+        $categoriesLinks = $this->getCategoriesLinks($crawler);
+        foreach($categoriesLinks as $item) {
+            $response = $client->request('GET', $item);
+            $crawler = new Crawler($response->getBody()->getContents(), 'http');
```
Comment (tfe2012) [inline src/ImportBundle/Command/GetLinksCommand.php]: This 2 lines must have dependencies from interface, not from class

File: src/ImportBundle/Command/GetLinksCommand.php [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+        $id = $input->getArgument('id');
+
+        $service = $this->getContainer()->get('import.link.parser');
+        $service2 = $this->getContainer()->get('import.product.link');
+
+        $shopLink = $service->start($id);
+
+        $client = new GuzzleHttp\Client();
+        $response = $client->request('GET', $shopLink);
+        $crawler = new Crawler($response->getBody()->getContents(), 'http');
```
Comment (tfe2012) [inline src/ImportBundle/Command/GetLinksCommand.php]: This 2 lines must have dependencies from interface, not from class

File: src/ImportBundle/Command/GetLinksCommand.php [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+        foreach($categoriesLinks as $item) {
+            $response = $client->request('GET', $item);
+            $crawler = new Crawler($response->getBody()->getContents(), 'http');
+            $pagination = $this->getPagesCount($crawler);
+            for ($i = 1; $i <= $pagination; $i++) {
+                $response = $client->request('GET', $item . '/?page=' . $i);
+                $crawler = new Crawler($response->getBody()->getContents(), 'http');
+                $products = $this->getCategoryProducts($crawler);
+                foreach ($products as $item) {
+                    $message = $service2->insertShopProductsLinks($id, $item);
```
Comment (tfe2012) [inline src/ImportBundle/Command/GetLinksCommand.php]: For product info parsing, we have other command

**Pull request merged**